### PR TITLE
Implement conversation muting

### PR DESCRIPTION
**Pull request by aapeliv**
**Backend checklist**
- [x] Formatted my code by running `autoflake -r -i --remove-all-unused-imports src && isort . && black .` in `app/backend`
- [x] Added tests for any new code or added a regression test if fixing a bug
- [ ] All tests pass
- [ ] Run the backend locally and it works
- [ ] Added migrations if there are any database changes, rebased onto `develop` if necessary for linear migration history

### CODE REVIEW

**Review by kalvdans — DISMISSED** [review]
LGTM! Although a bit non-symmetrical to not also have a dummy timestamp in the past for the unmuted case.

**Comment by aapeliv** [comment]
> LGTM! Although a bit non-symmetrical to not also have a dummy timestamp in the past for the unmuted case.

That's a good point, it'll probably simplify if done that way...

**Comment by aapeliv** [comment]
By the way @kalvdans, are you able to join the backend meeting in ~2h? Details at https://couchers.org/contributors/backend-meeting.ics

**Comment by aapeliv** [comment]
> LGTM! Although a bit non-symmetrical to not also have a dummy timestamp in the past for the unmuted case.

Fixed!

File: app/proto/conversations.proto [hunk 1]
```diff
@@ -133,6 +138,12 @@ message Message {
   }
 }
 
+message MuteInfo {
+  bool muted = 1;
+  // muted_until is not present if muted forever
+  google.protobuf.Timestamp muted_until = 2;
```
Comment (kalvdans) [inline app/proto/conversations.proto:145]: Perhaps it is easier to pass a duration here. "mute for the next 8 hours". Then we don't need the error code for muting in the past, and timezones are less of a trouble.

**Review by kalvdans — DISMISSED** [review]
LGTM with anther remark

File: app/proto/conversations.proto (hunk 1 above)
Comment (aapeliv) [reply app/proto/conversations.proto:145]: Good point!

Comment (aapeliv) [reply app/proto/conversations.proto:145]: Fixed

Comment (kalvdans) [reply app/proto/conversations.proto:145]: Hmm, we didn't get rid of the error code anyway since Duration is signed :( But still better than before IMHO!

**Review by kalvdans — APPROVED** [review]

File: app/proto/conversations.proto (hunk 1 above)
Comment (aapeliv) [reply app/proto/conversations.proto:145]: Yes, and it's better for the frontend

**Pull request merged**